osu.Game/Screens/Select/BeatmapDetails.cs

```diff
@@ -331,7 +331,13 @@ public string Text
                 {
                     if (string.IsNullOrEmpty(value))
                     {
-                        this.FadeOut(transition_duration);
+                        if (Height > 0)
+                        {
+                            this.FadeOut(transition_duration);
+                            return;
+                        }
+
+                        Hide();
```

Comment:
I don't think this is a proper fix, either. It's pretty hacky to say the least.

The core problem here is that the metadata section is initially visible while it should probably not be. I think the way this should be fixed is by setting `Alpha = 0` in the ctor of `MetadataSection` (but *not* moving it out of the nested `FillFlowContainer` - see discussion in #3632 for the reason why).

Keeping that above PR in mind (this is an ages-old regression, it seems), I'd be much more comfortable with this change in general if it had visual test cases attached.